parser-typechecker/src/Unison/Runtime/IOSource.hs

```diff
@@ -570,12 +570,15 @@ unique[de2e0ee924578939213c950dfd8e0ba1047703ae] type ANSI.Color
   | BrightBlack | BrightRed | BrightGreen | BrightYellow | BrightBlue
   | BrightMagenta | BrightCyan | BrightWhite
 
-List.map : (a ->{g} b) ->{} [a] ->{g} [b]
-List.map f as =
-  go acc = cases
-    [] -> acc
-    h +: t -> go (acc :+ f h) t
-  go [] as
+List.map : (a ->{e} b) -> [a] ->{e} [b]
+List.map f a =
+  go i as acc =
+    match List.at i as with
+      None   -> acc
+      Some a ->
+        use Nat +
+        go (i + 1) as (acc :+ f a)
```

Comment:
Kinda interesting that `List.map` in base uses indexing rather than repeated uncons. The idea is to avoid performance hit of allocation and rebalancing while traversing, and my guess is that pays off for the right sequence type, but I wonder if it's true for Data.Sequence, the current implementation of lists.

Anyway, if/when List.map changes in base, we can swap this out.